pkg/daemon/ceph/osd/volume.go

```diff
@@ -1129,9 +1129,11 @@ func GetCephVolumeRawOSDs(context *clusterd.Context, clusterInfo *client.Cluster
 			}
 
 			// Close encrypted device
-			err = CloseEncryptedDevice(context, block)
-			if err != nil {
-				return nil, errors.Wrapf(err, "failed to close encrypted device %q for osd %d", block, osdID)
+			if block != "" {
```

Comment:
Should the encrypted device only be closed if `isDeviceEncrypted` is true as well?

Reply:
at this point, we already know that the device is encrypted. Rook uses `CEPH_VOLUME_DMCRYPT_SECRET` env variable in line 1117 above.